src/algorithms/bin_path_info.cpp

```diff
@@ -71,10 +71,13 @@ void bin_path_info(const PathHandleGraph& graph,
                         }
                         bins[curr_bin].mean_pos += path_pos++;
                         nucleotide_count += 1;
-                        if(bins[curr_bin].first_nucleotide == 0){
-                            bins[curr_bin].first_nucleotide = nucleotide_count;
-                        }
-                        bins[curr_bin].last_nucleotide = nucleotide_count;
+						if((bins[curr_bin].ranges.size() == 0) ||
+						    ((nucleotide_count -  bins[curr_bin].ranges[bins[curr_bin].ranges.size() - 1].second) > 1)){
```

Comment:
https://en.cppreference.com/w/cpp/container/vector/back

Reply:
I was not aware of this, will change accordingly.